src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -168,6 +168,11 @@ def validate_client_parameters(cmd, namespace):
     if n.sas_token:
         n.sas_token = n.sas_token.lstrip('?')
 
+    # account name with secondary
+    if n.account_name.endswith('-secondary'):
```

Comment:
When will this logic be hit?

Reply:
when customer specify --account-name as `xxx-secondary`